Library/AppEnvironment.swift

```diff
@@ -142,10 +143,12 @@ public struct AppEnvironment: AppEnvironmentType {
   }
 
   internal static func resetStackForUnitTests() {
-    while self.stack.popLast() != nil {}
+    while self.stack.count > 1 {
+      _ = self.stack.popLast()
+    }
 
     let next = Environment()
-    self.pushEnvironment(next)
+    self.replaceCurrentEnvironment(next)
```

Comment:
This fixes a crash in `resetStackForUnitTests` because it will never be truly empty. If it's empty, calls to `isFeatureEnabled` will fail.